src/Dialog/formComponents/validator/versionConstraintFormat.js

```diff
@@ -0,0 +1,25 @@
+export default function versionContraintFormat(data) {
+  const semVerFormat = new RegExp([
+    /v?([0-9]+)(\.[0-9]+)?/.source,
+    /(\.[0-9]+)?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?/.source,
+    /(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?$/.source
+    ].join('')
+  );
```

Comment:
@p-kozlowski Please check regexp

Reply:
I remember somewhere else in the codebase we went for https://github.com/semver/semver/issues/232#issuecomment-405596809 which is a bit different

Reply:
OK, I'll handle this.

Reply:
@p-kozlowski done :)